### ISSUE TITLE
Session is lost / chatroom closed on page refresh (F5) or browser navigation

### ISSUE THREAD

**Issue by benjamin-dk** [cfdp/opeka#48]
When the user refreshes the page or uses browser navigation - back / forward - the session is lost and the user has to log back in.

This is true for all user roles.

The user should at least have a popup warning that the session will be lost.

The best solution would be a standard popup warning is shown with a wording similar to "Do you really want to leave this page?" [Stay] [Leave].

**Comment by alex-bukach** [comment]
> The user should at *least* have a popup warning that the session will be lost.
@benjamin-dk so should we prevent loosing the session, or should we only notify user that the session will be lost?

**Comment by cfdp** [comment]
@alex-bukach if the user chooses to stay on the page, the session should *not* be lost. This seems to me to be the most logical solution.

**Comment by alex-bukach** [comment]
@benjamin-dk I'm stuck with this line: https://github.com/cfdp/opeka/blob/master/nodejs/static/connect.js#L7563. It causes loosing the session on "beforeunload", i.e. simultaneously with showing the "Do you really want to leave this page?" popup (so the session is lost even if a user selects to stay). Therefore I'm inclined to remove that line and find another workaround. Why might we need to run the chat in an iframe?

**Comment by benjamin-dk** [comment]
@alex-bukach we never run the chat itself in an iframe, as far as I know.

We do, however most of the time initiate a chat session via an embedded iframe such as these https://demo.curachat.com/widget-demo.

When a user clicks the "Open chat" button a new window/tab opens and the chat login form is presented.

If your workaround can leave this feature (and the remaning functionality) intact feel free to remove the iframe related code.

**Comment by alex-bukach** [comment]
@benjamin-dk That line serves to disconnect chat on "beforeunload" event (since "unload" event is mentioned not to work in Opera in iframe).

I have commented out that line, then created a page containing the chat in the iframe, and tested it in the latest version of Opera. As nodejs log shows, the chat is shut down as expected, i.e. "unload" event works properly there.

I have found mentions of issues with unload and beforeunload events in both Opera and other browsers, however they are quite old, so this issue seems to be resolved in Opera. May I ask you to test https://github.com/cfdp/opeka/pull/73 and confirm whether it works on your side as well.
